policy/reviewer/reviewer.go

```diff
@@ -124,31 +127,94 @@ func FindRandomRequesters(ctx context.Context, prctx pull.Context, result common
 			shoveIntoMap(allUsers, orgMembers)
 		}
 
-		allCollaboratorPermissions, err := prctx.RepositoryCollaborators()
+		allCollaboratorPermissions := make(map[string]string)
+		allCollaboratorAndPermissions, err := prctx.RepositoryCollaborators()
 		if err != nil {
 			return nil, errors.Wrap(err, "Unable to list repository collaborators")
 		}
 
-		if child.ReviewRequestRule.Admins {
-			var repoAdmins []string
-			for user, perm := range allCollaboratorPermissions {
-				if perm == common.GithubAdminPermission {
-					repoAdmins = append(repoAdmins, user)
-				}
-			}
-			shoveIntoMap(allUsers, repoAdmins)
-		}
-
 		if child.ReviewRequestRule.WriteCollaborators {
 			var repoCollaborators []string
-			for user, perm := range allCollaboratorPermissions {
+			for user, perm := range allCollaboratorAndPermissions {
 				if perm == common.GithubWritePermission {
 					repoCollaborators = append(repoCollaborators, user)
 				}
 			}
 			shoveIntoMap(allUsers, repoCollaborators)
 		}
 
+		// When admins are selected for review, only collect the desired set of admins instead of
+		// everyone, which includes org admins
+		if !child.ReviewRequestRule.Admins {
+			// When not looking for admins, we want to check with everyone
+			allCollaboratorPermissions = allCollaboratorAndPermissions
+		} else {
+			// Determine what the scope of requested admins should be
+			switch child.ReviewRequestRule.AdminScope {
+			case common.User:
+				logger.Debug().Msg("Selecting admin users with direct collaboration rights")
+				directCollaborators, err := prctx.DirectRepositoryCollaborators()
+				if err != nil {
+					logger.Error().Err(err).Msgf("Unable to get list of direct collaborators on %s", prctx.RepositoryName())
```

Comment:
Seeing these error logs that continue on with execution is surprising (even if it works); I think this is another reason to make a helper function.